examples/cloudml-document-ai-patents/automl_ner.py

```diff
@@ -0,0 +1,146 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Runs AutoML NER on the text and writes results to BigQuery."""
+
+import logging
+import os
+import utils
+
+from google.cloud import storage, bigquery
+from google.cloud import automl_v1beta1 as automl
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def extract_field_from_payload(text, payload, field_name, default_value='None'):
+  """Parses a payload to extract the value of the given field.
+
+  Args
+    text: text analyzed by AutoML NER.
+    payload: payload returned by AutoML NER.
+    field_name: Name of the field to extract.
+    default_value: Value to return if the field can not be found.
+
+  Returns:
+    extracted value.
+
+  In case the payload contains several times the given field, we take the occurence
+    with the highest score.
+  """
+  value_found = default_value
+  score_found = -1
+  for result in payload:
+    extracted_field_name = result.display_name
+    extracted_value_start = result.text_extraction.text_segment.start_offset
+    extracted_value_end = result.text_extraction.text_segment.end_offset
+    
+    extracted_value = text[extracted_value_start:extracted_value_end]
+
+    score = result.text_extraction.score
+    
+    if (extracted_field_name == field_name) and (score > score_found):
+      score_found = score
+      value_found = extracted_value
+  return value_found
+
+
+def run_automl_single(ocr_path,
+                      list_fields,
+                      service_acct,
+                      model_id,
+                      main_project_id,
+                      compute_region):
+  """Runs AutoML NER on a single document and returns the dictionary of results."""
+
+  # Set up client for AutoML NER model
+  automl_client = automl.AutoMlClient.from_service_account_json(service_acct)
+  model_full_id = automl_client.model_path(
+      main_project_id, compute_region, model_id)
+  prediction_client = automl.PredictionServiceClient.from_service_account_json(service_acct)
+
+  # Load text
+  text = utils.download_string(ocr_path, service_acct).read().decode('utf-8')
+
+  # Call AutoML
+  payload = {"text_snippet": {"content": text, "mime_type": "text/plain"}}
+  params = {}
+  response = prediction_client.predict(model_full_id, payload, params)
+  
+  # Parse results
+  results = {'file': os.path.basename(ocr_path).replace('.txt', '.pdf')}
+  for field in list_fields:      
+      value_field = extract_field_from_payload(text, response.payload, field)
+      results[field] = value_field
+  return results
+
+def predict(main_project_id,
+            input_path,
+            demo_dataset,
+            demo_table,
+            model_id,
+            service_acct,
+            compute_region,
+            config):
+  """Runs AutoML NER on a folder and writes results to BigQuery.
```

Comment:
Docstring looks out of date vs the parameters and is missing the longer description.